ext/scripts/install_scripts/install_via_r_remotes.pl

```diff
@@ -54,13 +54,57 @@ =head1 SYNOPSIS
 }
 
 
-my $combining_template = "library(remotes)\n<<<<0>>>>";
+my $combining_template = '
+library(remotes)
+install_or_fail <- function(package_name, version){
+
+   tryCatch({install_version(package_name, version, repos="https://cloud.r-project.org", Ncpus=4)
+         library(package_name, character.only = TRUE)},
+         error = function(e){
+             print(e)
+             stop(paste("installation failed for:",package_name ))},
+         warning = function(w){
+           catch <-
+             grepl("download of package .* failed", w$message) ||
+             grepl("(dependenc|package).*(is|are) not available", w$message) ||
+             grepl("installation of package.*had non-zero exit status", w$message) ||
+             grepl("installation of one or more packages failed", w$message)
+           if(catch){ print(w$message)
+             stop(paste("installation failed for:",package_name ))}}
+         )
+
+ }
+
+<<<<0>>>>
+';
+
+my $combining_template_validation = '
+
+available_packages <- available.packages()
+
+validate_or_fail <- function(package_name){
+    # Check if the package is in the list of available packages
+    is_installed <- package_name %in% rownames(available_packages)
+
+    # Check the result
+    if (!is_installed) {
+        stop(paste("Package nor installed:", package_name))
+    }
+}
+
+
+<<<<0>>>>
+';
+
+
 my @separators = ("\n");
-my @templates = ('install_version("<<<<0>>>>",NULL,repos="https://cloud.r-project.org", Ncpus=4)');
+my @templates = ('install_or_fail("<<<<0>>>>",NULL)');
```

Comment:
Maybe let's rename it to install template

Reply:
Done